src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java

```diff
@@ -138,22 +138,32 @@ public class XpathRegressionTest extends AbstractModuleTestSupport {
             "SuppressWarningsHolder"
     );
 
-    private static Set<String> simpleCheckNames;
-    private static Map<String, String> allowedDirectoryAndChecks;
-    private static Set<String> internalModules;
+    private static final Set<String> simpleCheckNames = getSimpleCheckNames();
+    private static final Map<String, String> allowedDirectoryAndChecks =
+        getAllowedDirectoryAndChecks();
+
+    private static final Set<String> internalModules = getInternalModules();
```

Comment:
Can we move initialization of these into the `BeforeEach` below (as well as content of methods that initialize them)?

Reply:
We can, but why? Initializing `internalModules` doesn't throw an exception and initializing `simpleCheckNames` and `allowedDirectoryAndChecks` initially shouldn't cause any problem.

Reply:
If this just initializes the values once, and they never change during test runs, then it doesn't make sense to run them in `BeforeEach` as it will just generate the same lists every time. `BeforeEach` should really be for areas that need to be reset or prepare for each run.

Reply:
They don't change during the tests.

Reply:
> If this just initializes the values once, and they never change during test runs, then it doesn't make sense to run them in `BeforeEach` as it will just generate the same lists every time. `BeforeEach` should really be for areas that need to be reset or prepare for each run.

Agreed, let’s not do this.

Reply:
`PMD Failure: com.puppycrawl.tools.checkstyle.internal.XpathRegressionTest:154 Rule:AvoidThrowingRawExceptionTypes Priority:1 Avoid throwing raw exception types..`

Should I create a custom exception?

Reply:
No, please avoid custom exceptions at all costs.

Reply:
I added it to the suppression list.